bk-api/bk_api.py

```diff
@@ -860,34 +878,36 @@ def deploy_wes(node_id, this_debug, force=False):
 
     return {"success":True}
 
-def add_vsn_event(node_id,field_value, lock_tables=True, lock_requested_by=""):
+def add_vsn(node_id):
+    proxy = get_cluster_node_subprocess_proxy(node_id)
 
-    payload = {"node_id": node_id, "source": "beekeeper-add-vsn", "operation":"insert",
-                "field_name": "vsn", "field_value": field_value}
+    output = proxy.check_output(["cat", "/etc/waggle/vsn"], text=True)
+    vsn = output.strip()
```

Comment:
did we lost the .upper() when collecting the VSN from the nodes?

Reply:
This is a good question.

After reading it, I actually added a validation step to make sure it's uppercase letters and numbers only.

We can force uppercase it or we can decide that it's up to us to make sure /etc/waggle/vsn is the correct format.

Reply:
okay if it fails cause its lower case, how do we know?  what is gonna notify us?  will registration fail?  or will it just be a fail on the beekeeper logs?

Reply:
For future notes - we discussed this and the main point is, during WSN build and registration, we already have some validation and correction when a node's VSN is entered. Assuming this is sufficient, no new failures should occur during registration.

It is still the team's responsibility to be aware of this behavior and be able to check the beekeeper logs if nodes are not correctly deploying WES.

Reply:
For the format of VSN, I think this was the verdict when we had this conversation: https://github.com/waggle-sensor/beekeeper/blob/main/bk-api/bk_api.py#L882